### ISSUE TITLE
[CRASH] Changing location of imported files crashes Olive on project opening

### ISSUE THREAD

**Issue by FirasH** [olive-editor/olive#1260]
**Commit Hash** 
0.2.0 ac4be090

**Platform**
Windows 10

**Summary**
If the location of the imported files is changed in a Project, Olive crashes opeining the project.

**Steps to Reproduce**

1. Launch Olive and go to: `File > Import...` and import a file (same with all formats)
2. Save the Project
3. Change the location of the imported file in step 1

Olive crashes consistently.

<details><summary><strong>Backtrace</strong></summary><pre><code>

</code></pre></details>

**Additional Information**

**Comment by FirasH** [comment]
Forgot to mention that I found this issue in the project I'm working on right now.
When I delete a specific `Math` node Olive crashes.

In order to report that issue, I changed the imported files location to see if I could reproduce the issue without the need to upload everything.

Anyway I'll keep a copy of the current project frozen for a future report (if still repriducible).

**Comment by ThomasWilshaw** [comment]
Thanks, this is a known issue. If the math node problem is a separate problem can you open a new issue, if not can you provide more details? Thanks

**Comment by FirasH** [comment]
> Thanks, this is a known issue. If the math node problem is a separate problem can you open a new issue, if not can you provide more details? Thanks

I’ll try to simplify as much as possible the project to isolate the issue and report another issue with a link.

**Comment by FirasH** [comment]
@ThomasWilshaw, just created a separate report for the `Math Node` issue (#1261).

**Comment by ThomasWilshaw** [comment]
So I think the actual crash here is also fixed by https://github.com/olive-editor/olive/pull/1262 as the crash is the `SampleBuffer` trying to allocate bad data. However I guess it would be good to warn the user earlier if a file can't be opened. Video streams fail quietly already without a crash. @FirasH if you import just an image (i.e. no audio source) and follow your steps, do you get a crash or the following warning in the console:
```
[WARNING] Failed to open decoder for "C:/Users/Tom/Desktop/movetest/new_rrt_32.p
ng" :: 0 (class std::shared_ptr<class olive::Decoder> __cdecl olive::RenderWorke
r::ResolveDecoderFromInput(class std::shared_ptr<class olive::Stream>):360)
```
Or similar

**Comment by ThomasWilshaw** [comment]
This is most likely fixed in the `otio` branch which will hopefully be merged soon.

**Comment by FirasH** [comment]
Great!
Keep up the good work guys!

**Comment by itsmattkc** [comment]
Should be fixed as of db8d321fa713ad2c5979df637a3f9990363539e7
